### PR TITLE
lightdm-mini-greeter: init at 0.3.2

### PR DESCRIPTION
**Pull request by mnacamura**
###### Motivation for this change

[lightdm-mini-greeter](https://github.com/prikhi/lightdm-mini-greeter) is a neat, minimalistic LightDM greeter. It could be alternative to SLiM.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
@emmanuelrosa For turning off xterm you might want to set the respective option to `false`: https://nixos.org/nixos/options.html#xterm.

**Comment by mnacamura** [comment]
@emmanuelrosa You can choose your session in `~/.dmrc`. Otherwise you can configure the default desktop session in `/etc/nixos/configuration.nix` by for example
```
services.xserver.desktopManager.default = "none";
services.xserver.windowManager.default = "bspwm";
```
in my case using bspwm.

**Comment by emmanuelrosa** [comment]
This is what it took to make it work with `i3`:
```
services.xserver.windowManager.default = "i3";
services.xserver.desktopManager.xterm.enable = false;
```

When I set the Desktop Manager to "none" `i3` started but so did `xterm`, and killing `xterm` also killed `i3`.

Given this, the only recommendation I have is to document what is needed to set the window/desktop manager (since this greeter doesn't provide a way to do so via the GUI). Better yet, an option which simply takes care of it. Something like this:

`services.xserver.displayManager.lightdm.greeters.mini.userSession = "i3";`

**Comment by mnacamura** [comment]
Ahh I was wrong. `~/.dmrc` is ignored. [Mini greeter seems to start only the default session.](https://github.com/prikhi/lightdm-mini-greeter/blob/a28ff60ace80b9e8804bfbf370393ded290e8d02/src/callbacks.c#L12)

> Given this, the only recommendation I have is to document what is needed to set the window/desktop manager (since this greeter doesn't provide a way to do so via the GUI). Better yet, an option which simply takes care of it. Something like this:
> 
> services.xserver.displayManager.lightdm.greeters.mini.userSession = "i3";

To document something like ``For using this greeter, you need to set the default session by ...'' would be sufficiently helpful IMHO. 

> This is what it took to make it work with i3:
> 
> services.xserver.windowManager.default = "i3";
> services.xserver.desktopManager.xterm.enable = false;

I tested
```
services.xserver.desktopManager.default = "none";
services.xserver.windowManager.default = "i3";
```
and it worked fine. I did not need to set `services.xserver.desktopManager.xterm.enable = false`.

**Comment by mnacamura** [comment]
All set, maybe. I'm waiting for review.

**Comment by mnacamura** [comment]
All set, maybe. I'm waiting for review.

File: nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix
Comment (matthewbauer) [inline nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix]: This add a few too many options. Can add an "extraConfig" like the lightdm-gtk-greeter has? Any options in common with lightdm-gtk-greeter are probably worth keeping.

File: pkgs/applications/display-managers/lightdm-mini-greeter/default.nix
Comment (matthewbauer) [inline pkgs/applications/display-managers/lightdm-mini-greeter/default.nix]: If you could add yourself as maintainer, this becomes a little easier to merge.

**Review by matthewbauer — CHANGES_REQUESTED** [review]

File: nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix
Comment (mnacamura) [reply nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix]: I agree. However, mini-greeter requires that in the config file, all the options are explicitly specified; it does not provide default options when they are omitted in the config file. So, unlike GTK greeter, it could be unsafe to introduce `extraConfig`, with which users might forget to specify some options and after `nixos-rebuild switch` they cannot login.

Comment (mnacamura) [reply nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix]: I added `extraConfig` and most options can be set there.

Comment (mnacamura) [reply nixos/modules/services/x11/display-managers/lightdm-greeters/mini.nix]: @matthewbauer Could you review this again?

**Pull request merged**
